### DISCUSSION TITLE
How many squares, of any size, are on a board divided up into 6 by 6 squares?

### DISCUSSION THREAD

**Discussion by ParanoidUser** [ParanoidUser/yolo#711] [Galaxy Trivia] [answered]
![image](https://user-images.githubusercontent.com/5120290/236932556-55868725-134c-472e-9412-35987de55424.png)

**Comment by peterbaccaro** [comment]
There is 1 square of size 6x6.
There are 4 squares of size 5x5.
There are 9 squares of size 4x4.
There are 16 squares of size 3x3.
There are 25 squares of size 2x2.
There are 36 squares of size 1x1.

Therefore, the total number of squares, of any size, on a board divided up into 6 by 6 squares is:

1 + 4 + 9 + 16 + 25 + 36 = **91**.

**Comment by thuongtruong1009** [comment, marked as answer]
To determine the number of squares on a 6x6 board, we need to consider squares of all possible sizes. Let's break it down:

1x1 squares: There are 6x6 = 36 1x1 squares on the board.

2x2 squares: There are (6-1) x (6-1) = 5x5 = 25 2x2 squares on the board.

3x3 squares: There are (6-2) x (6-2) = 4x4 = 16 3x3 squares on the board.

4x4 squares: There are (6-3) x (6-3) = 3x3 = 9 4x4 squares on the board.

5x5 squares: There are (6-4) x (6-4) = 2x2 = 4 5x5 squares on the board.

6x6 squares: There is only 1 6x6 square on the board.

Adding these all up, we get:

1x1 squares: 36
2x2 squares: 25
3x3 squares: 16
4x4 squares: 9
5x5 squares: 4
6x6 squares: 1

The total number of squares on the 6x6 board is 36 + 25 + 16 + 9 + 4 + 1 = 91 squares.

**Comment by ParanoidUser** [comment]
**Bonus:** What is the mathematical formula for calculating the total subsquares in a square of arbitrary size?

**Reply by peterbaccaro** [reply]
The formula for calculating the total number of sub-squares in a square matrix of squares is given by:

**N = (n * (n + 1) * (2n + 1)) / 6**

where "n" is the number of squares along each edge of the matrix.

To understand this formula, we can consider that for a matrix of size "n x n", there are "n-1" smaller squares along each edge, "n-2" squares of size 2x2, "n-3" squares of size 3x3, and so on, down to 1 square of size "n-n". The total number of squares is the sum of all these smaller squares. This sum can be expressed as a cubic polynomial in "n", which can be simplified using algebra to yield the above formula.

For example, for a 3x3 matrix of squares, we have "n=3", and the formula gives:

N = (3 * 4 * 7) / 6 = 14

This means that there are a total of 14 sub-squares in the 3x3 matrix.
